### DISCUSSION TITLE
Compare type ExifTag accepts against ExifDataType

### DISCUSSION THREAD

**Discussion by prabhavmehra** [SixLabors/ImageSharp#2575] [Q&A] [answered]
Is it possible to compare the type ExifTag accepts against an ExifDataType?

The main workflow I am looking into -
A custom `IDictionary<ExifTag, object>` which the app can use to set the values.
For example, `CustomDict[ExifTag.GPSLatitude] = 144`. And then in my custom implementation I want to check which type does GPSLatitude accepts (Rational[]) in this case and then convert the object to that type and set the exif tag.

This mainly simplifies the code the app has where it doesn't has to bother about types that much and the custom implementation does everything.

One thing I thought was to have a dictionary or similar to hold the tag and what type it accepts but that is difficult to manage based on the number of tag,

**Comment by prabhavmehra** [comment, marked as answer]
```c#
Type tagType = ExifTag.{tag}.GetType();

// switch (tagType)
case Type t when t == typeof(ExifTag<byte>): // change `byte` to any other type
```

**Reply by JimBobSquarePants** [reply]
I guess you could do that...

I'm struggling to understand what case you are trying to solve here tbh. Our `ExifProfile` type is strongly typed.

**Reply by prabhavmehra** [reply]
Mainly allowing the app to not bother about type conversion and some other functionality. 

Here are few scenarios:
- If the app has longitude and latitude and in float, double etc. The app shouldn't bother about it being Rational[] and type conversion. It should be able to set it to custom implementation and that should take care of type conversion and set it accordingly.
- Similar goes for other types (for example Timestamp, where I can have a datetimeoffset but the app can just set that with custom implementation and the conversion is handled by the library).
- If setting longitude and latitude, I want to set their respective ref as well (which is not done by default) based on if they are above/below 0.

@JimBobSquarePants

**Reply by JimBobSquarePants** [reply]
That sound like something you'd plug into converters or utility methods.

**Reply by prabhavmehra** [reply]
Does ImageSharp provide any similar convertors? Also, what would this look like (a high-level overview? something like a IValueConvertor? Method Extensions?)

The implementation I have currently does the same. i.e. provides some sort of convertor to convert the 'x' value type to the type the ExifTag accepts. It's just done when you set it in the custom IDictionary. 

another I reason I went for this approach is to have some functionality like data binding, custom events (as the custom Dictionary implements INotifyPropertyChanged,

@JimBobSquarePants
